crates/net/common/src/bandwidth_meter.rs

```diff
@@ -69,16 +84,74 @@ impl BandwidthMeter {
     pub fn total_outbound(&self) -> u64 {
         self.inner.outbound.load(Ordering::Relaxed)
     }
+
+    pub fn ingress_throughput(&self) -> f64 {
+        f64::from_bits(self.inner.ingress_throughput.load(Ordering::Relaxed))
+    }
+
+    pub fn egress_throughput(&self) -> f64 {
+        f64::from_bits(self.inner.egress_throughput.load(Ordering::Relaxed))
+    }
 }
 
 impl Default for BandwidthMeter {
     fn default() -> Self {
-        Self {
+        let bandwidth_meter = Self {
             inner: Arc::new(BandwidthMeterInner {
                 inbound: AtomicU64::new(0),
+                ingress_delta: AtomicU64::new(0),
+                ingress_throughput: AtomicU64::new(0),
                 outbound: AtomicU64::new(0),
+                egress_delta: AtomicU64::new(0),
+                egress_throughput: AtomicU64::new(0),
+                throughput_period: Duration::from_secs(1),
             }),
+        };
+
+        tokio::spawn(ThroughputTask::new(
+            bandwidth_meter.inner.throughput_period,
+            bandwidth_meter.clone(),
+        ));
+
+        bandwidth_meter
+    }
+}
+
+/// An endless Future used to calculate ingress & egress throughputs
+/// for the given [`BandiwdthMeter`] on the given [`Interval`].
```

Comment:
If we agree with having this background task, then I would like to expand it to have one task handle updating throughput for *all* `BandwidthMeter`s, instead of one background task per meter